spinalcordtoolbox/scripts/sct_analyze_lesion.py

```diff
@@ -888,19 +928,112 @@ def angle_correction(self):
-                                                                  slice_number=self.midsagittal_sc_slice_rpi,
-                                                                  axis=0)  # 0 = RL
-        printv(f'Midsagittal slice of the spinal cord: {self.midsagittal_sc_slice}', self.verbose, type='info')
+        # 1. Find lesion center of mass in S-I axis (z direction)
+        z_center = int(round(center_of_mass(im_lesion_data_largest_lesion)[2]))   # [2] --> S-I
+        # 2. Define analysis range (2 axial slices above and below the lesion center of mass) around lesion center
+        # mass in S-I axis (z direction)
+        # TODO: try other number of slices above and below the lesion center of mass
+        z_range = np.arange(z_center - 2, z_center + 3)   # 5 slices in total
+        # 3: For each of these slices, compute the spinal cord center of mass in the x-axis (R-L direction)
+        stored_x_coordinates = []
+        for z in z_range:
+            spinal_cord_slice = im_sc_data[:, :, z]     # RPI --> selecting in the 3rd dimension (SI) to get axial slice
+            if np.any(spinal_cord_slice):  # Avoid empty slices
+                stored_x_coordinates.append(center_of_mass(spinal_cord_slice)[0])   # [0] --> R-L
+        # 4. Calculate target position in right-left axis (x direction) for the interpolation (mean of spinal cord
+        # center of mass (in the x-axis (R-L direction))
+        self.interpolated_midsagittal_slice = np.mean(stored_x_coordinates)    # e.g., for [8.6, 8.6, 8.9, 8.8, 9.6] --> 8.7
+        printv(f'Interpolated midsagittal slice (same across lesions) = '
+               f'{round(self.interpolated_midsagittal_slice, 2)}', self.verbose, 'info')
+        # 5. Interpolate the lesion
+        slice1 = int(np.floor(self.interpolated_midsagittal_slice))     # e.g., 8
+        slice2 = int(np.ceil(self.interpolated_midsagittal_slice))      # e.g., 9
+        self.interpolation_slices = [slice1, slice2]     # store it to be used for tissue bridges
+
+    def _interpolate_values(self, data1, data2):
+        """
+        Interpolate inputs using linear interpolation.
+        Inputs can be either 2D numpy arrays (for interpolating slices) or single int64 (for interpolating tissue
+        bridges).
+        :param data1: 2D numpy array (slice 1) or single int64 (tissue bridge for slice 1)
+        :param data2: 2D numpy array (slice 2) or single int64 (tissue bridge for slice 2)
+        :return: 2D numpy array (interpolated slice) or single float64 (interpolated tissue bridge)
+        """
+        interpolation_factor = self.interpolated_midsagittal_slice - int(self.interpolated_midsagittal_slice)   # e.g., 8.7 - 8 = 0.7
+        return (1 - interpolation_factor) * data1 + interpolation_factor * data2
+
+    def _interpolate_lesion(self, im_lesion_data):
+        """
+        Interpolate the currently processed lesion
```

Comment:
I found this method name to be a bit confusing? To me, it's not clear what "interpolate_lesion" actually means...

Maybe a name like `_get_lesion_midsagittal_slice` could be a bit more descriptive? (That way, it matches the naming convention of `_compute_width_midsagittal_slice` and `_compute_length_midsagittal_slice`.)


```suggestion
    def _get_lesion_midsagittal_slice(self, im_lesion_data):
        """
        Get the lesion data from the midsagittal slice by interpolating between the
        the slices above and below.
```

Reply:
Sounds good; the method renamed in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4702/commits/e131faaa8ee230021508fd81a432fd5ab7df057c

> interpolating between the slices above and below.

just a minor clarification here: we interpolate two sagittal slices (R-L axis), not the slices above and below. I clarified this in the method docstring